kong/db/schema/init.lua

```diff
@@ -1644,57 +1661,131 @@ function Schema:process_auto_fields(data, context, nulls, opts)
-          data[key] = utils.uuid()
+        if is_insert_or_upsert and value == nil then
+          value = uuid()
         end
 
-      elseif field.type == "string" then
-        if is_insert_or_upsert and data[key] == nil then
-          data[key] = utils.random_string()
+      elseif ftype == "string" then
+        if is_insert_or_upsert and value == nil then
+          value = random_string()
         end
 
-      elseif (key == "created_at" and is_insert_or_upsert and (data[key] == null or
-                                                               data[key] == nil))
+      elseif (key == "created_at" and is_insert_or_upsert and (value == null or
+                                                               value == nil))
       or
              (key == "updated_at" and (is_insert_or_upsert or context == "update"))
       then
-        if field.type == "number" then
+        if ftype == "number" then
           if not now_ms then
             update_time()
             now_ms = ngx_now()
           end
-          data[key] = now_ms
+          value = now_ms
 
-        elseif field.type == "integer" then
+        elseif ftype == "integer" then
           if not now_s then
             update_time()
             now_s = ngx_time()
           end
-          data[key] = now_s
+          value = now_s
         end
       end
     end
 
-    data[key] = adjust_field_for_context(field, data[key], context, nulls, opts)
+    value = adjust_field_for_context(field, value, context, nulls, opts)
 
     if is_select then
-      if data[key] == null and not nulls then
-        data[key] = nil
-      elseif field.type == "integer" and type(data[key]) == "number" then
-        data[key] = floor(data[key])
+      local vtype = type(value)
+      if value == null and not nulls then
+        value = nil
+      elseif ftype == "integer" and vtype == "number" then
+        value = floor(value)
+      end
+
+      if resolve_references then
+        --local ref
+        if ftype == "string" and field.referencable then
+          if is_reference(value) then
+            --ref = value
+            local deref, err = dereference(value)
+            if deref then
+              value = deref
+            else
+              if err then
+                kong.log.warn("unable to resolve reference ", value, "(", err, ")")
+              else
+                kong.log.warn("unable to resolve reference ", value)
+              end
+            end
+          end
+
+        elseif vtype == "table" and (ftype == "array" or ftype == "set") then
+          local subfield = field.elements
+          if subfield.type == "string" and subfield.referencable then
+            local count = #value
+            if count > 0 then
+              for i = 1, count do
+                if is_reference(value[i]) then
+                  --if not ref then
+                  --  ref = { value }
+                  --else
+                  --  ref[i] = value
+                  --end
```

Comment:
Let's add sentence as to why this is commented out for now

Reply:
It is on separate commit. Perhaps we should just remove the whole code for now. Just thinking what we put in beta.